core/ecschema-metadata/src/UnitProvider/SchemaUnitProvider.ts

```diff
@@ -149,55 +168,72 @@ export class SchemaUnitProvider implements UnitsProvider {
    * @returns The UnitConversionProps interface from the @itwin/core-quantity package.
    */
   public async getConversion(fromUnit: UnitProps, toUnit: UnitProps): Promise<UnitConversionProps> {
-    const conversion = await this._unitConverter.calculateConversion(fromUnit.name, toUnit.name);
-    return {
+    // need to check if either side is an inverted unit. The UnitConverter can only handle Units
+    if (!fromUnit.isValid || !toUnit.isValid)
+      throw new BentleyError(BentleyStatus.ERROR, "Both provided units must be valid.", () => {
+        return { fromUnit, toUnit };
+      });
+
+    const { unitName: fromUnitName, isInverted: fromIsInverted } = await this.checkUnitPropsForConversion(fromUnit, this._context);
+    const { unitName: toUnitName, isInverted: toIsInverted } = await this.checkUnitPropsForConversion(toUnit, this._context);
+
+    const conversion = await this._unitConverter.calculateConversion(fromUnitName, toUnitName);
+    const result: UnitConversionProps = {
       factor: conversion.factor,
       offset: conversion.offset,
     };
+    if (fromIsInverted && !toIsInverted)
+      result.inversion = UnitConversionInvert.InvertPreConversion;
+    else if (!fromIsInverted && toIsInverted)
+      result.inversion = UnitConversionInvert.InvertPostConversion;
+
+    return result;
   }
 
-  /**
-   * Find unit in a schema that has unitName.
-   * @param unitName Full name of unit.
-   * @returns Unit whose full name matches unitName.
-   */
-  private async findECUnitByName(unitName: string): Promise<Unit> {
-    // Check if schema exists and unit exists in schema
-    const [schemaName, schemaItemName] = SchemaItem.parseFullName(unitName);
+  private async checkUnitPropsForConversion(input: UnitProps, context: SchemaContext): Promise<{ unitName: string, isInverted: boolean }> {
+    const [schemaName, schemaItemName] = SchemaItem.parseFullName(input.name);
     const schemaKey = new SchemaKey(schemaName);
-    const schema = await this._context.getSchema(schemaKey);
+    const schema = await context.getSchema(schemaKey);
 
     if (!schema) {
-      throw new BentleyError(BentleyStatus.ERROR, "Cannot find schema for unit", () => {
-        return { schema: schemaName, unit: unitName };
+      throw new BentleyError(BentleyStatus.ERROR, "Could not obtain schema for unit.", () => {
+        return { name: input.name };
       });
     }
 
     const itemKey = new SchemaItemKey(schemaItemName, schema.schemaKey);
-    const item = await this._context.getSchemaItem<Unit>(itemKey);
-    if (!item)
-      throw new BentleyError(BentleyStatus.ERROR, "Cannot find schema item/unit", () => {
-        return { item: schemaItemName, schema: schemaName };
-      });
-
-    if (item.schemaItemType === SchemaItemType.Unit)
-      return item;
+    const invertedUnit = await context.getSchemaItem<InvertedUnit>(itemKey);
+    if (invertedUnit && InvertedUnit.isInvertedUnit(invertedUnit) && invertedUnit.invertsUnit) {
```

Comment:
Hmm, could the 2 latter checks here not be reduced into 1?

Reply:
Actually, can an InvertedUnit schema item exist, but the unit it is inverting does not exist? I guess that can explain the need for the 3rd check, but if not, we could instead check invertedUnit.invertsUnit in this new `isInvertedUnit()` method no?

Reply:
the comment does help yep. This should be fine as is then